### PR TITLE
deps: update all sorts of stuff

### PR DESCRIPTION
**Pull request by ckipp01**
This pr goes through and makes the various updates:

Bumps mill to the latest 0.10.5
Bumps to the latest Scala versions for 2 and latest 3.1.x version
Move Scala native to 0.4.x
Adds in dependabot
Updates checkout and java actions

### CODE REVIEW

**Comment by ckipp01** [comment]
I'm having some issues with the `WatchTests`, but even locally on master they fail for me so I want to try them in CI here.

**Review by lefou — COMMENTED** [review]
From glancing over this PR, I can't spot the change where you remove the Scala.js 0.x support. I'm general fine with this PR except the Scala.js 0.6 part, and I was about asking, if you can split the PR to leave that change out. But now, I even don't see it. Do I need a rest? 🤔  Can you point me to it?

**Comment by ckipp01** [comment]
> From glancing over this PR, I can't spot the change where you remove the Scala.js 0.x support.

Ah, sorry, this was laziness on my part. I've been updating a bunch of repos in the org lately and I copied the commit message over from another one without thinking since this has no scala js modules. I'll update the commit.

**Comment by ckipp01** [comment]
> except the Scala.js 0.6 part

Note that in many of the other com-lihaoyi repos I did remove support for 0.6 though.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: 2
+updates:
+
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "monthly"
```
Comment (lefou) [inline .github/dependabot.yml:7]: I'd prefer a weekly check.

File: build.sc [hunk 2]
```diff
@@ -72,12 +68,11 @@ object os extends Module {
 }
 
 trait OsLibModule extends CrossScalaModule with PublishModule{
-  def isDotty = crossScalaVersion.startsWith("0") || crossScalaVersion.startsWith("3")
   def publishVersion = VcsVersion.vcsState().format()
   def pomSettings = PomSettings(
     description = artifactName(),
     organization = "com.lihaoyi",
-    url = "https://github.com/lihaoyi/os",
+    url = "https://github.com/com-lihaoyi/os-lib",
```
Comment (lefou) [inline build.sc:75]: Nice spot! :+1:

**Comment by lefou** [comment]
> > except the Scala.js 0.6 part
> 
> Note that in many of the other com-lihaoyi repos I did remove support for 0.6 though.

Yeah, I know. We just need to make sure, we bump the major version (of 0.x releases, this is semantically equivalent to the minor version).

File: .github/dependabot.yml (hunk 1 above)
Comment (ckipp01) [reply .github/dependabot.yml]: Done.

**Comment by lefou** [comment]
Thank you!

**Pull request merged**